src/hooks/v2/contractReader/ETHReceivedFromTokens.ts

```diff
@@ -0,0 +1,61 @@
+import { BigNumber } from '@ethersproject/bignumber'
+import { BallotState } from 'models/ballot-state'
+import { useContext } from 'react'
+import { parseWad, permyriadToPercent } from 'utils/formatNumber'
+import { V2ProjectContext } from 'contexts/v2/projectContext'
+
+import useProjectReservedTokens from './ProjectReservedTokens'
+
+// Returns the value in ETH that an amount of tokens can be redeemed for
+export function useETHReceivedFromTokens({
+  tokenAmount,
+}: {
+  tokenAmount: string | undefined
+}) {
+  const {
+    projectId,
+    fundingCycle,
+    fundingCycleMetadata,
+    overflow,
+    totalTokenSupply,
+    ballotState,
+  } = useContext(V2ProjectContext)
+
+  const { data: undistributedReservedTokens } = useProjectReservedTokens({
+    projectId,
+    reservedRate: fundingCycleMetadata?.reservedRate,
+  })
+
+  if (!fundingCycle) return
+
+  const realTotalTokenSupply = totalTokenSupply?.add(
+    undistributedReservedTokens ?? BigNumber.from(0),
+  )
+
+  if (!fundingCycleMetadata || !realTotalTokenSupply?.gt(0))
+    return BigNumber.from(0)
+
+  const redemptionRate =
+    ballotState === BallotState.Active
+      ? permyriadToPercent(fundingCycleMetadata.ballotRedemptionRate)
+      : permyriadToPercent(fundingCycleMetadata.redemptionRate)
+
+  const redemptionRateBN = BigNumber.from(redemptionRate)
+
+  const base =
```

Comment:
Naming is hard but I have no idea what `base` means here. Is there a better name?

Reply:
fuck it couldn't come up with something better, added a comment lol